Describe how to view Elastic Agent status (#884)
elastic/observability-docs: docs/en/ingest-management/elastic-agent/view-elastic-agent-status.asciidoc

[[view-elastic-agent-status]]
[role="xpack"]
= View status of {agent}s

To view the status of your {fleet}-managed agents, go to
*Management > Fleet > Agents*.

[role="screenshot"]
image::images/agent-status.png[Agents tab showing status of each {agent}]

{fleet} reports these statuses:

|=== 

| *Healthy* | The {agent} is online and running as expected. There are no agent

> policy updates or automatic agent binary updates pending, but there might be

Reviewer: There are no agent policy updates or automatic agent binary updates ** in progress**

> policy updates or automatic agent binary updates pending, but there might be

Author: ```suggestion
policy updates or automatic agent binary updates in progress, but there might be
```

> manual agent binary updates available.

Reviewer: but the agent binary may still be out of date.

> policy updates or automatic agent binary updates pending, but there might be
> manual agent binary updates available.

Author: ```suggestion
policy updates or automatic agent binary updates in progress, but the
agent binary may still be out of date.
```

| *Unhealthy* | The {agent} is online, but requires attention because it's
reporting either agent or integration-level errors.

| *Updating* | The {agent} is online and updating the agent policy, updating the
binary, or enrolling or unenrolling from {fleet}.

| *Offline* | The {agent} has not checked in during the expected time period.

> | *Inactive* | The {agent} is inactive.

Author: ```suggestion
| *Inactive* | The {agent} is inactive no longer enrolled in {fleet}. The agent must re-enroll to connect to {fleet} again.
```

|===

> //QUESTION: What elase can we say about the offline status? That the agent

Author: ```suggestion
//QUESTION: What else can we say about the offline status? That the agent
```

//might be stopped or no longer able to connect to Fleet? What else?

//QUESTION: What does inactive mean? It shows up in the dropdown.

//QUESTION: Can users set alerts on these statuses yet? If so, we should mention

> //that here.

Author: @mostlyjason Can you respond to these questions? I can merge as-is but thought it would be good to add more detail if we can.

Reviewer: Inactive means the agent is no longer enrolled in Fleet. This is a terminal state and agents must reenroll if they wish to connect to Fleet again.

Reviewer: I'm not aware of any way to set an alert.  I don't think people would want to because presumably they unenrolled the agent because they are done with it.

> //QUESTION: What else can we say about the offline status? That the agent
> //might be stopped or no longer able to connect to Fleet? What else?
> 
> //QUESTION: What does inactive mean? It shows up in the dropdown.
> 
> //QUESTION: Can users set alerts on these statuses yet? If so, we should mention
> //that here.
> 

Author: ```suggestion
```

[discrete]
[[view-agent-list]]
== Filter the list of agents

To filter the list of agents by status, click the *Status* dropdown and select
one or more statuses.

[role="screenshot"]
image::images/agent-status-filter.png[Agent Status dropdown with mutliple statuses selected]

For advanced filtering, use the search bar to create structured queries
using {kibana-ref}/kuery-query.html[Kibana Query Language]. For example, enter
`local_metadata.os.family : "darwin"` to see only agents running on macOS.

[discrete]
[[view-agent-logs]]
== View agent logs

When {fleet} reports a status like `Offline` or `Unhealthy`, you might want to
view the agent logs to diagnose potential causes. If logging is configured for
the agent:

. Under *Agents*, click the agent name listed in the *Host* column.

. Click the *Logs* tab.
+

> On the *Logs* tab you can filter, search, and explore the agent logs.

Reviewer: Mention that they can remotely turn debug logging on here too by changing the log level?

> On the *Logs* tab you can filter, search, and explore the agent logs.

Author: ```suggestion
On the *Logs* tab you can filter, search, and explore the agent logs.
You can also remotely turn on debug logging by changing the log level.
```

+
[role="screenshot"]
image::images/view-agent-logs.png[View agent logs under agent details]
+
To tail the agent log files in real time, click *Open in Logs*.

For more information about logging, refer to:

> //TODO: uncomment after logging topic is merged.
> //* <<elastic-agent-logging>>

Author: ```suggestion
* <<elastic-agent-logging>>
```
